micro-rdk/src/esp32/pwm.rs

```diff
@@ -219,14 +219,27 @@ impl<'a> LedcTimerWrapper<'a> {
         if self.count > 0 {
             return Err(Esp32PwmError::OtherChannelsBindToTimer);
         }
+
         let id = {
             let timer = self.timer.take();
             timer.unwrap().timer() as u8
         };
         let timer_config = TimerConfig::default().frequency(frequency_hz.Hz());
-        let _ = self.timer.set(create_timer_driver(id, &timer_config)?);
-        self.frequency = frequency_hz;
-        Ok(())
+
+        match create_timer_driver(id, &timer_config) {
+            Ok(driver) => {
+                let _ = self.timer.set(driver);
+                self.frequency = frequency_hz;
+                Ok(())
+            }
+            Err(err) => {
+                // if there is a failure, or the frequency is bad, we want to revert back
+                // to the last working frequency before raising the error
+                let timer_config = TimerConfig::default().frequency(self.frequency.Hz());
+                let _ = self.timer.set(create_timer_driver(id, &timer_config)?);
```

Comment:
What should happen if this restoring call to `create_timer_driver` *also* fails. Right now it will return the error from trying and failing to restore the timer, and not the error from the attempt to create the timer with the new frequency. It will also leave the timer in what I believe is still an invalid state.

Reply:
So I actually don't believe this is possible, because in order for an `LedcTimerWrapper` to exist, this needs to have been called with a proper frequency at some point prior (because of the code in `new`).

Reply:
OK. I'd say `?` isn't the right thing then. If this is believed to be known to be impossible (by us), we should probably panic! if if happens. So, maybe `expect`?

Reply:
yeah this is reasonable, I'll change it to an expect